### PR TITLE
feat: add tag and category table

### PR DESCRIPTION
**Pull request by mpfan**
This pr adds all the necessary tables.

### CODE REVIEW

File: src/main/resources/db/migration/V002__Shopify_tables.sql [hunk 1]
```diff
@@ -12,4 +12,15 @@ create table migrations.user (
...
+
+create table migrations.shop (
+  id BIGINT GENERATED ALWAYS AS IDENTITY,
+  owner_id BIGINT,
+  name text NOT NULL,
+  description TEXT,
+  PRIMARY KEY (id),
+  CONSTRAINT fd_owner_id
+    FOREIGN KEY (owner_id)
+      REFERENCES migrations.user(id)
```
Comment (mpfan) [inline src/main/resources/db/migration/V002__Shopify_tables.sql:25]: Try this: `shop_owner_id bigint references migrations.user`

File: src/main/resources/db/migration/V002__Shopify_tables.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+);
+
+create table migrations.product(
+    id bigint generated always as identity primary key,
+    shop_id BIGINT REFERENCES migrations.shop(id),
+    name text not null,
+    price BIGINT not null,
```
Comment (mpfan) [inline src/main/resources/db/migration/V002__Shopify_tables.sql:32]: use type `decimal` for price

File: src/main/resources/db/migration/V002__Shopify_tables.sql [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    quantity BIGINT not null,
```
Comment (mpfan) [inline src/main/resources/db/migration/V002__Shopify_tables.sql:33]: use type `integer`

**Review by dershao — APPROVED** [review]
excellent

**Review by DwangDistracted — APPROVED** [review]

**Pull request merged**